### ISSUE TITLE
Dependent roles run multiple time

### ISSUE THREAD

**Issue by sorki** [nestihacky/nestihacky-everything#1]
If two roles has a role in their deps the dependent role runs twice, this forces us to place common configuration into group_vars/host_vars file.

**Comment by mmilata** [comment]
Weird, the [docs](http://docs.ansible.com/ansible/playbooks_roles.html#role-dependencies) say:

> Roles dependencies are always executed before the role that includes them, and are recursive. By default, roles can also only be added as a dependency once - if another role also lists it as a dependency it will not be run again. This behavior can be overridden by adding allow_duplicates: yes to the meta/main.yml file.

**Comment by sorki** [comment]
Yeah, I also found bunch of issues discussing this. It's not designed to handle such cases and needs workarounds like used by debops https://docs.debops.org/en/latest/debops-policy/docs/code-standards-policy.html#debops-policy-ref-code-standards-dependent-variables

Ime gonna apply a bit of haskell glue instead.
